### PR TITLE
feat: add SocialMediaSharing component

### PR DESCRIPTION
**Pull request by christian-dinh**
### Overview

Ports the [SocialMediaSharing](https://github.com/codecademy-engineering/Codecademy/tree/develop/webpack/assets/portal/components/SocialMediaSharing) component over from the monolith. Offloads tracking to the consumer via an `action` prop, similar to [GlobalHeader](https://github.com/Codecademy/client-modules/blob/main/packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx#L32)

Notes:
- The `small` label text size is 14 instead of 12 since the latter is no longer valid per our styleguide, but I found no usages that had both a `small` and a `label` prop
- Per conversation with Li I have dropped the hover styles for the `white` version of the component as
  - it used the [deprecated](https://gamut.codecademy.com/storybook/?path=/docs/foundations-colors--page#deprecated-colors) version of `green-400`
  - the `black` version has no hover styles anyway. The `white` prop is only passed in the [CheatSheetHeader](https://www.codecademy.com/learn/learn-html/modules/learn-html-elements/cheatsheet) and `/learner-stories/:story-type/:id` which I can't figure out how to get to

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [REACH-982](https://codecademy.atlassian.net/browse/REACH-982)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.13-alpha.6de40d.0<br/>@codecademy/gamut-labs@19.1.0-alpha.6de40d.0<br/>@codecademy/styleguide@41.3.0-alpha.6de40d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60bf99dfec5aa2008f444b04--gamut-preview.netlify.app>https://60bf99dfec5aa2008f444b04--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60bf99dfec5aa2008f444b04>Deploy Logs</a></p>

File: packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  small,
+  track,
+  sectionId,
+  white,
+}) => {
+  const onClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    track?.(e);
+    const left = (screen.width - 570) / 2; // eslint-disable-line no-restricted-globals
+    const top = (screen.height - 570) / 2; // eslint-disable-line no-restricted-globals
```
Comment (christian-dinh) [inline packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:35]: `screen` is considered a restricted global in `client-modules`, is there a reason why?

https://caniuse.com/mdn-api_screen

File: packages/gamut-labs/src/SocialMediaSharing/index.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    formatShare: ({ url }: SocialMediaShare) => ({ url }),
+  },
+];
+
+type SocialMediaSharingProps = {
+  url: string;
+  message?: string;
+  hashtags?: string[];
+  mention?: string;
+  action?: (e: React.MouseEvent, target: string) => void;
```
Comment (MattSchiller) [inline packages/gamut-labs/src/SocialMediaSharing/index.tsx:70]: nit: Is this actually `track`? Or would it be better to call this a more genericized `onClick`? NBD though, naming is subjective, so if this reads better to you, ignore me =)

File: packages/gamut-labs/src/SocialMediaSharing/index.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  label?: string;
+  small?: boolean;
+  sectionId?: string;
+  white?: boolean;
+  iconStyles?: string;
+};
```
Comment (MattSchiller) [inline packages/gamut-labs/src/SocialMediaSharing/index.tsx:76]: So it looks like a lot of these props are just being marshaled down to SocialShareIconLink?

If that's the case, I'd recommend you pull the shared props into their own type in the SocialShareIconLink file and have both of these components' props include them in their prop type. That way:
- It's obvious that's what these props are
- It informs our compiler that they're the same, so if you go to this shared type and right click a key and "find all references" it'll find the key all up the DOM tree for easier debugging/refactoring
- It'll make it easier to add/remove these items in the future

**Review by MattSchiller — APPROVED** [review]
Looks good, though I have a few comments that I think would really help tie the code together.

But no real blockers 👍

File: packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+`.withComponent(Anchor);
+
+type SocialShareIconLinkProps = {
+  id: string;
+  icon: React.ComponentType<GamutIconProps>;
+  href: string;
+  small?: boolean;
+  track?: (e: React.MouseEvent) => void;
+  sectionId?: string;
+  white?: boolean;
```
[the remark this answers is not in the document]
Comment (christian-dinh) [reply packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:19]: how do you feel about `variant='black'|'white'` with black being the default?

File: packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:35]: Ha, if you want to disable `no-restricted-globals` in `client-modules` I wouldn't be opposed. It's not super duper useful now that we have TypeScript.

File: packages/gamut-labs/src/SocialMediaSharing/index.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+// https://developers.facebook.com/docs/sharing/reference/share-dialog
+// https://docs.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/share-on-linkedin?context=linkedin/consumer/context#create-an-article-or-url-share
+export const SOCIAL_SHARING_PLATFORMS = [
+  {
+    id: 'facebook',
+    icon: FacebookIcon,
+    baseUrl: 'https://www.facebook.com/dialog/share?app_id=212500508799908',
+    formatShare: ({ url, message }: SocialMediaShare) => ({
+      ...(url && { href: url }),
+      ...(message && { quote: message }),
```
[the remark this answers is not in the document]
Comment (christian-dinh) [reply packages/gamut-labs/src/SocialMediaSharing/index.tsx:42]: cmd+c / cmd+v baybee 😎

File: packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:19]: Ha! I was _about_ to type the same thing. +1 to descriptor names like `size` and `variant` over boolean names like `small` and `white`.

File: packages/gamut-labs/src/SocialMediaSharing/index.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+import { SocialShareIconLink } from './SocialShareIconLink';
+
+export type SocialMediaShare = {
+  url: string;
+  message?: string;
+  hashtags?: string[];
+  mention?: string;
+};
+
+export const shareLinkCreator = (
```
[the remark this answers is not in the document]
Comment (christian-dinh) [reply packages/gamut-labs/src/SocialMediaSharing/index.tsx:18]: this was a copy/paste (along with some better colocation) and I don't think it actually needs to be partially applied like this at all lol

File: packages/gamut-labs/src/SocialMediaSharing/index.tsx (hunk 3 above)
Comment (christian-dinh) [reply packages/gamut-labs/src/SocialMediaSharing/index.tsx:76]: does a `Pick` work, or do we lose out on some of those benefits?

Screen Shot 2021-06-08 at 3 52 26 PM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.13-alpha.0eef75.0<br/>@codecademy/gamut-labs@19.1.0-alpha.0eef75.0<br/>@codecademy/styleguide@41.3.0-alpha.0eef75.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60bfcfc3fefbcd52b10bbe3c--gamut-preview.netlify.app>https://60bfcfc3fefbcd52b10bbe3c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60bfcfc3fefbcd52b10bbe3c>Deploy Logs</a></p>

File: packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (MattSchiller) [reply packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx]: image

File: packages/gamut-labs/src/SocialMediaSharing/index.tsx (hunk 3 above)
Comment (christian-dinh) [reply packages/gamut-labs/src/SocialMediaSharing/index.tsx]: gotcha, good point on circular deps btw (there is [tooling](https://www.npmjs.com/package/madge) we can add to the PR builder someday!)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.13-alpha.160695.0<br/>@codecademy/gamut-labs@19.1.0-alpha.160695.0<br/>@codecademy/styleguide@41.3.0-alpha.160695.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60bfe0b809c1dd00e4b25ac7--gamut-preview.netlify.app>https://60bfe0b809c1dd00e4b25ac7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60bfe0b809c1dd00e4b25ac7>Deploy Logs</a></p>

File: packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+const IconLink = styled(Box)`
+  &:last-of-type {
+    margin-right: 0;
+  }
+`.withComponent(Anchor);
+
+export type BaseSocialShareProps = {
+  sectionId?: string;
+  size?: 'small' | 'normal';
+  variant?: 'black' | 'white';
```
[the remark this answers is not in the document]
Comment (christian-dinh) [reply packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:15]: I can def use variants for `size`. Only thing with color modes is that the white version of this component is used on [cheatsheet pages](https://www.codecademy.com/learn/introduction-to-javascript/modules/learn-javascript-introduction/cheatsheet) which use legacy styles, would there be a way to override the color mode in cases like that?

[the remark this answers is not in the document]
Comment (christian-dinh) [reply packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:15]: oh wait `ButtonBase` takes a `mode` prop, I don't 100% know how color modes work I guess but does that mean that it's only taken from context if a parent has a color mode set or something, and the `mode` prop can override?

[the remark this answers is not in the document]
Comment (christian-dinh) [reply packages/gamut-labs/src/SocialMediaSharing/SocialShareIconLink.tsx:15]: alternatively am I totally making up the fact that it ever comes from context? lol

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.13-alpha.d67b06.0<br/>@codecademy/gamut-labs@19.1.0-alpha.d67b06.0<br/>@codecademy/styleguide@41.3.0-alpha.d67b06.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60c1354b9db65e0f9e7b9d9a--gamut-preview.netlify.app>https://60c1354b9db65e0f9e7b9d9a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60c1354b9db65e0f9e7b9d9a>Deploy Logs</a></p>

**Comment by christian-dinh** [comment]
per slack convo w @codecaaron
>I'd say ship it

and so it shall be!

**Pull request merged**